frontend/src/components/authchooser/index.tsx

```diff
@@ -0,0 +1,158 @@
+import { Box, Button, DialogTitle, withStyles } from '@material-ui/core';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { generatePath, useHistory, useLocation } from 'react-router-dom';
+import { useClustersConf } from '../../lib/k8s';
+import { request, testAuth } from '../../lib/k8s/apiProxy';
+import { createRouteURL } from '../../lib/router';
+import { getCluster, getClusterPrefixedPath } from '../../lib/util';
+import { setConfig } from '../../redux/actions/actions';
+import { ClusterDialog } from '../cluster/Chooser';
+import { Loader } from '../common';
+import OauthPopup from '../oidcauth/OauthPopup';
+
+const ColorButton = withStyles((theme) => ({
+  root: {
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.sidebarBg,
+    width: '14rem',
+    padding: '0.5rem 2rem',
+    '&:hover': {
+      opacity: '0.8',
+      backgroundColor: theme.palette.sidebarBg
+    }
+  },
+}))(Button);
+
+interface ReactRouterLocationStateIface {
+  from?: Location;
+}
+
+function AuthChooser(){
+  const history = useHistory();
+  const location = useLocation();
+  const clusters = useClustersConf();
+  const dispatch = useDispatch();
+  const [testingAuth, setTestingAuth] = React.useState(false);
+  const {from = { pathname: createRouteURL('cluster') }} = (location.state || {}) as ReactRouterLocationStateIface;
+  const clusterName = getCluster() as string;
+
+  let clusterAuthType:string = '';
+  if (clusters[clusterName]){
+    clusterAuthType = clusters[clusterName].auth_type;
+  }
+
+  function tokenAuthHandler(){
+    history.push({
+      pathname: generatePath(getClusterPrefixedPath('token'), {cluster: clusterName as string}),
+    });
+  }
+
+  function oidcAuthHandler(){
+    history.replace({
+      pathname: generatePath(getClusterPrefixedPath(), {cluster: clusterName as string}),
+    });
+  }
+  function oauthWindowCloseHandler(){
+    //gets called when oauth window popup is closed
+  }
+
+  React.useEffect(() => {
+    request(`registerselectedauthcluster?cluster=${clusterName}`, {}, false, false);
```

Comment:
As mentioned, we should not have this step as it may lead to issues if multiple users and multiple clusters are being used at the same time.